packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -100,20 +101,32 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
       return onUpdate
     }
 
+    // Load texture helper waits for the texture and image to finish loading
+    // before returning.
+    async[$loadTexture](uri: string) {
+      let result: Texture|null = null;
+      await new Promise(resolve => {
+        return new TextureLoader().load(uri, texture => {
+          resolve(texture);
```

Comment:
This is very close to what `TextureUtils.load()` does. Think you can use that instead, perhaps with some light modification?

Reply:
It'll save you from creating and destroying `TextureLoader`s all the time, which I think is discouraged.

Reply:
Ok updated to do something similar.